beacon-chain/state/state-native/beacon_state_mainnet.go

```diff
@@ -99,23 +107,46 @@ type beaconStateMarshalable struct {
 }
 
 func (b *BeaconState) MarshalJSON() ([]byte, error) {
+	var bRoots customtypes.BlockRoots
+	var sRoots customtypes.StateRoots
+	var mixes customtypes.RandaoMixes
+	var balances []uint64
+	var inactivityScores []uint64
+	var vals []*ethpb.Validator
+
+	if features.Get().EnableExperimentalState {
+		bRoots = b.blockRootsMultiValue.Value(b)
+		sRoots = b.stateRootsMultiValue.Value(b)
+		mixes = b.randaoMixesMultiValue.Value(b)
+		balances = b.balancesMultiValue.Value(b)
+		inactivityScores = b.inactivityScoresMultiValue.Value(b)
+		vals = b.validatorsMultiValue.Value(b)
+	} else {
+		bRoots = b.blockRoots
+		sRoots = b.stateRoots
+		mixes = b.randaoMixes
+		balances = b.balances
+		inactivityScores = b.inactivityScores
+		vals = b.validators
+	}
+
 	marshalable := &beaconStateMarshalable{
 		Version:                             b.version,
 		GenesisTime:                         b.genesisTime,
 		GenesisValidatorsRoot:               b.genesisValidatorsRoot,
 		Slot:                                b.slot,
 		Fork:                                b.fork,
 		LatestBlockHeader:                   b.latestBlockHeader,
-		BlockRoots:                          b.blockRoots,
-		StateRoots:                          b.stateRoots,
+		BlockRoots:                          &bRoots,
```

Comment:
Is there a reason to have the Marshable state? can we avoid the blockroots being pointers there? It seems backward and error prone to have slices on one structure and pointers on the other.

Reply:
I don't know what is the exact purpose of the marshalable state, but I removed pointers